src/schneider_lab_to_nwb/schneider_2024/schneider_2024_behaviorinterface.py

```diff
@@ -1,25 +1,109 @@
+                name = event_dict["name"]
+                times = np.array(file["events"][name]["time"]).squeeze()
+                name_to_times[name] = times
+            for event_dict in metadata["Behavior"]["ValuedEvents"]:
+                name = event_dict["name"]
+                times = np.array(file["events"][name]["time"]).squeeze()
+                values = np.array(file["events"][name]["value"]).squeeze()
+                name_to_times[name] = times
+                name_to_values[name] = values
+
+        # Add Data to NWBFile
+        behavior_module = nwb_helpers.get_module(
+            nwbfile=nwbfile,
+            name="behavior",
+            description="Behavioral data from the experiment.",
+        )
+
+        # Add BehavioralTimeSeries
+        behavioral_time_series = BehavioralTimeSeries(
+            time_series=behavioral_time_series,
+            name="behavioral_time_series",
+        )
+        behavior_module.add(behavioral_time_series)
+
+        # Add Events
+        event_types_table = EventTypesTable(name="event_types", description="Metadata about event types.")
+        event_type_name_to_row = dict()
+        i = 0
+        for event_dict in metadata["Behavior"]["Events"]:
+            event_type_name_to_row[event_dict["name"]] = i
+            event_types_table.add_row(
+                event_name=event_dict["name"],
+                event_type_description=event_dict["description"],
+            )
+            i += 1
+        for event_dict in metadata["Behavior"]["ValuedEvents"]:
+            event_type_name_to_row[event_dict["name"]] = i
+            event_types_table.add_row(
+                event_name=event_dict["name"],
+                event_type_description=event_dict["description"],
+            )
+            i += 1
+        events_table = EventsTable(
+            name="events_table",
+            description="Metadata about events.",
+            target_tables={"event_type": event_types_table},
+        )
+        events_table.add_column(name="value", description="Value of the event.")
+        for event_dict in metadata["Behavior"]["Events"]:
+            event_times = name_to_times[event_dict["name"]]
+            event_type = event_type_name_to_row[event_dict["name"]]
+            for event_time in event_times:
+                events_table.add_row(timestamp=event_time, event_type=event_type, value="")
+        for event_dict in metadata["Behavior"]["ValuedEvents"]:
+            event_times = name_to_times[event_dict["name"]]
+            event_values = name_to_values[event_dict["name"]]
+            event_type = event_type_name_to_row[event_dict["name"]]
+            for event_time, event_value in zip(event_times, event_values):
+                events_table.add_row(timestamp=event_time, event_type=event_type, value=str(event_value))
+        behavior_module.add(events_table)
 
-        raise NotImplementedError()
+        task = Task(event_types=event_types_table)
```

Comment:
Is the event_types_table global?

Do you think it would make sense to have two EventsTable instead of one? That way for the non valued events you don't need to extra column.

Reply:
> Is the event_types_table global?

What do you mean?

Reply:
> Do you think it would make sense to have two EventsTable instead of one? That way for the non valued events you don't need to extra column.

Yeah I think that's reasonable.

Reply:
That there can be only even_types_table in the nwfile.